ocaml/xapi/storage_smapiv1_wrapper.ml

```diff
@@ -359,6 +359,25 @@ module Everything = struct
     Errors.errors := h.errors
 end
 
+let with_tracing ~name ~dbg f =
+  let dbg, parent =
+    match String.index_opt dbg '\x00' with
```

Comment:
Similar to the point in xenopsd, this operation here should not depend on an internal representation detail.
Please add a function in Context so this function can use something like: `Context.tracing_parent_of_dbg dbg`